ironfish-rust/src/transaction/tests.rs

```diff
@@ -2,44 +2,52 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 
-use std::collections::{BTreeMap, HashMap};
+use crate::{
+    errors::{IronfishError, IronfishErrorKind},
+    transaction::Transaction,
+};
 
-#[cfg(test)]
-use super::internal_batch_verify_transactions;
-use super::{ProposedTransaction, Transaction, TRANSACTION_PUBLIC_KEY_SIZE};
-use crate::frost_utils::account_keys::derive_account_keys;
-use crate::test_util::create_multisig_identities;
-use crate::transaction::tests::split_spender_key::split_spender_key;
+#[cfg(feature = "transaction-proofing")]
 use crate::{
     assets::{asset::Asset, asset_identifier::NATIVE_ASSET},
-    errors::{IronfishError, IronfishErrorKind},
-    frost_utils::split_spender_key,
+    frost_utils::{account_keys::derive_account_keys, split_spender_key::split_spender_key},
     keys::SaplingKey,
     merkle_note::NOTE_ENCRYPTION_MINER_KEYS,
     note::Note,
     sapling_bls12::SAPLING,
-    test_util::make_fake_witness,
+    test_util::{create_multisig_identities, make_fake_witness},
     transaction::{
-        batch_verify_transactions, verify_transaction, TransactionVersion,
-        TRANSACTION_EXPIRATION_SIZE, TRANSACTION_FEE_SIZE, TRANSACTION_SIGNATURE_SIZE,
+        verify::batch_verify_transactions, verify::internal_batch_verify_transactions,
+        verify_transaction, ProposedTransaction, TransactionVersion, TRANSACTION_EXPIRATION_SIZE,
+        TRANSACTION_FEE_SIZE, TRANSACTION_PUBLIC_KEY_SIZE, TRANSACTION_SIGNATURE_SIZE,
     },
 };
+#[cfg(feature = "transaction-proofing")]
 use ff::Field;
+#[cfg(feature = "transaction-proofing")]
 use group::GroupEncoding;
-use ironfish_frost::dkg::{round1 as round1_dkg, round2 as round2_dkg, round3 as round3_dkg};
-use ironfish_frost::participant::Secret;
+#[cfg(feature = "transaction-proofing")]
 use ironfish_frost::{
```

Comment:
Why is the frost stuff behind this flag? Is it just unused outside of proof context?

Reply:
Yes, it's what you said: the frost tests require creating new transactions, which requires the feature to be enabled. If the feature is disabled, the tests cannot run, and these imports are unused.